dev/tasks/conda-recipes/arrow-cpp/meta.yaml

```diff
@@ -1,6 +1,6 @@
-{% set version = "3.0.0" %}
+{% set version = ARROW_VERSION %}
```

Comment:
Maybe can add a comment here indicating that setting this to `ARROW_VERSION` is a difference compared to the upstream version?